File: build-tools/codeassist-builder-plugin/src/main/java/com/tyron/builder/model/JavaArtifact.java
package com.tyron.builder.model;

import org.jetbrains.annotations.Nullable;

import java.io.File;

/**
 * The information for a generated Java artifact.
 */
public interface JavaArtifact extends BaseArtifact {

    /** Path to the mockable platform jar generated for this {@link JavaArtifact}, if present. */
    @Nullable
    File getMockablePlatformJar();
}

File: build-tools/codeassist-builder-plugin/src/main/java/com/tyron/builder/model/v2/ide/JavaArtifact.kt
package com.tyron.builder.model.v2.ide

import com.tyron.builder.model.v2.AndroidModel
import java.io.File

/**
 * The information for a generated Java artifact.
 *
 * This artifact is for Java components inside an Android project, which is only unit tests
 * for now.
 *
 * @since 4.2
 */
interface JavaArtifact : AbstractArtifact, AndroidModel {
    /** Path to the mockable platform jar generated for this [JavaArtifact], if present.  */
    val mockablePlatformJar: File?

    /**
     * Returns the folder containing resource files that classes from this artifact expect to find
     * on the classpath.
     *
     * This is used to run the unit tests
     */
    val runtimeResourceFolder: File?
}
